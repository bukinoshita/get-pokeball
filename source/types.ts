export enum Pokeballs {
  Pokeball = 'pokeball',
  Greatball = 'greatball',
  Ultraball = 'ultraball',
  Safariball = 'safariball',
  Masterball = 'masterball',
}

export type PokeballName =
  | Pokeballs.Pokeball
  | Pokeballs.Greatball
  | Pokeballs.Ultraball
  | Pokeballs.Safariball
  | Pokeballs.Masterball
