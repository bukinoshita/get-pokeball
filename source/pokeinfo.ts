import { Pokeballs } from './types'

export const pokeInfo = {
  [Pokeballs.Pokeball]: {
    name: 'Pok√© ball',
    price: {
      buy: 200,
      sell: 100,
    },
    slug: 'pokeball',
    value: {
      f: 12,
      d: 255,
    },
  },
  [Pokeballs.Greatball]: {
    name: 'Great ball',
    price: {
      buy: 600,
      sell: 300,
    },
    slug: 'greatball',
    value: {
      f: 8,
      d: 200,
    },
  },
  [Pokeballs.Ultraball]: {
    name: 'Ultra ball',
    price: {
      buy: 1200,
      sell: 600,
    },
    slug: 'ultraball',
    value: {
      f: 12,
      d: 150,
    },
  },
  [Pokeballs.Safariball]: {
    name: 'Safari ball',
    price: {
      buy: 'cannot be bought',
      sell: 'cannot be sold',
    },
    slug: 'safariball',
    value: {
      f: 12,
      d: 150,
    },
  },
  [Pokeballs.Masterball]: {
    name: 'Master ball',
    price: {
      buy: 'cannot be bought',
      sell: 'cannot be sold',
    },
    slug: 'masterball',
  },
}
