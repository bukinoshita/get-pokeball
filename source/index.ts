import { PokeballName } from './types'
import { pokeInfo } from './pokeinfo'

export function getPokeball(name: PokeballName) {
  return pokeInfo[name]
}
