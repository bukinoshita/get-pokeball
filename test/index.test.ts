import { getPokeball } from '../source'
import { Pokeballs } from '../source/types'

test('get pokeball', () => {
  const pokeball = getPokeball(Pokeballs.Pokeball)
  expect(pokeball).toMatchSnapshot()
})

test('get greatball', () => {
  const pokeball = getPokeball(Pokeballs.Greatball)
  expect(pokeball).toMatchSnapshot()
})

test('get ultraball', () => {
  const pokeball = getPokeball(Pokeballs.Ultraball)
  expect(pokeball).toMatchSnapshot()
})

test('get safariball', () => {
  const pokeball = getPokeball(Pokeballs.Safariball)
  expect(pokeball).toMatchSnapshot()
})

test('get masterball', () => {
  const pokeball = getPokeball(Pokeballs.Masterball)
  expect(pokeball).toMatchSnapshot()
})
