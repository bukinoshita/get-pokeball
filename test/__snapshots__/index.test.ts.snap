// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get greatball 1`] = `
Object {
  "name": "Great ball",
  "price": Object {
    "buy": 600,
    "sell": 300,
  },
  "slug": "greatball",
  "value": Object {
    "d": 200,
    "f": 8,
  },
}
`;

exports[`get masterball 1`] = `
Object {
  "name": "Master ball",
  "price": Object {
    "buy": "cannot be bought",
    "sell": "cannot be sold",
  },
  "slug": "masterball",
  "value": Object {
    "d": 0,
    "f": 0,
  },
}
`;

exports[`get pokeball 1`] = `
Object {
  "name": "Pok√© ball",
  "price": Object {
    "buy": 200,
    "sell": 100,
  },
  "slug": "pokeball",
  "value": Object {
    "d": 255,
    "f": 12,
  },
}
`;

exports[`get safariball 1`] = `
Object {
  "name": "Safari ball",
  "price": Object {
    "buy": "cannot be bought",
    "sell": "cannot be sold",
  },
  "slug": "safariball",
  "value": Object {
    "d": 150,
    "f": 12,
  },
}
`;

exports[`get ultraball 1`] = `
Object {
  "name": "Ultra ball",
  "price": Object {
    "buy": 1200,
    "sell": 600,
  },
  "slug": "ultraball",
  "value": Object {
    "d": 150,
    "f": 12,
  },
}
`;
